analyses/npl/01_get_stations.md

```diff
@@ -0,0 +1,379 @@
+```python
+import os
+from pathlib import Path
+import requests
+import yaml
+
+import tabula
+import fiona
+import pandas as pd
+import geopandas as gpd
+from selenium import webdriver
+from selenium.webdriver.firefox.options import Options
+from bs4 import BeautifulSoup
+
+
+DATA_DIR = Path(os.environ["AA_DATA_DIR"]) / 'public/exploration'
+DATA_DIR_PRIVATE = Path(os.environ["AA_DATA_DIR"]) / 'private'
+
+
+GLOFAS_STATION_FILE = DATA_DIR / 'glb/glofas/station_list.pdf'
+GLOFAS_STATION_OUTPUT = DATA_DIR / 'npl/glofas/npl_glofas_stations.gpkg'
+
+# Be careful if you want to run this, it takes awhile
+# and you can easily reach max requests
+SCRAPE_GOV_STATIONS = False
+GOV_BASE_URL = f"http://www.hydrology.gov.np/#/basin"
+GOV_DIR =  DATA_DIR / 'npl/dhm'
+GOV_STATION_FILENAME = GOV_DIR / 'npl_dhm_hydrology_stations.gpkg'
+GOV_BASIN_FILENAME = GOV_DIR / 'npl_dhm_hydrology_basins.csv'
+GOV_BASIN_SHAPEFILE = GOV_DIR / 'Nepal_Basin_final.kml'
+
+# Another station list    
+DHM_BASE_URL = "http://www.dhm.gov.np/hydrological-station/"
+DHM_STATION_FILENAME = GOV_DIR / 'npl_dhm_stations.gpkg'
+
+# Station list to share
+STATION_OUTPUT = GOV_DIR / 'station_list_glofas_final.xlsx'
+STATION_OUTPUT_YAML = GOV_DIR / 'station_list_glofas_final.yml'
+
+# From Ragindra
+RCO_DIR = DATA_DIR_PRIVATE / 'exploration/npl/unrco'
+BASINS_SHAPEFILE = RCO_DIR / 'shapefiles/Major_River_Basins.shp'
+
+# Final stations file
+STATIONS_FINAL = {
+    'glofas': DATA_DIR / 'npl/glofas/npl_glofas_stations_final.gpkg',
+    'dhm': GOV_DIR / 'npl_dhm_stations_final.gpkg'
+}
+```
+
+## Read in Nepal GloFAS stations
+
+```python
+tables = tabula.read_pdf(GLOFAS_STATION_FILE, pages=[51,52], 
+                         pandas_options=dict(
+                             header=None))
+
+```
+
+```python
+cnames = [ 
+          'station_name', 
+          'river_name', 
+          'river_basin',
+          'glofas_id',
+           'provider',
+          'provider_id',
+          'lat', 
+          'lon']
+
+cnums_lat_lon = {
+    0: [12, 14],
+    1: [13, 15]
+}
+
+table_list = []
+for i in range(2):
+    table = tables[i]
+    # Select only nepal
+    table = table[table[0] == 'Nepal Asia']
+    # Select only neeeded columns
+    table = table[[1, 2, 3, 4, 5, 7] + cnums_lat_lon[i]]
+    table.columns = cnames
+    table = gpd.GeoDataFrame(
+        table, 
+        geometry=gpd.points_from_xy(
+            table['lon'], table['lat']))
+    table_list.append(table)
+    
+df_glofas = pd.concat(table_list, ignore_index=True).set_crs("EPSG:4326")
```

Comment:
Generally, probably best to pull the crs from a global variable

Reply:
very good point, done!